### PR TITLE
journald: enforce longer line length limit during "setup" phase of st…

### PR DESCRIPTION
**Pull request by syyhao1994**
…ream protocol

This PR made modification on Lennart Poettering's basis. Fix the LineMax's function failure problem.

Fixes: #18993
Replaces: #19008， #19018

Signed-off-by: Yangyang Shen <shenyangyang4@huawei.com>

### CODE REVIEW

**Comment by syyhao1994** [comment]
@poettering PTAL

**Comment by syyhao1994** [comment]
With the PR https://github.com/systemd/systemd/pull/19337
``` 
-- Logs begin at Fri 2021-04-16 20:14:55 CST, end at Fri 2021-04-16 20:23:31 CST. --
Apr 16 20:20:59 localhost.localdomain systemd[1]: Starting test journald line max...
Apr 16 20:20:59 localhost.localdomain bash[5174]: asdkfjasasdfjasdlfajssdjfsdjfskldfjsljfdklsjdflksjdflksjdflsjdflksjdflksdjflsdj
Apr 16 20:20:59 localhost.localdomain bash[5174]: flsdjfasdjfalsjdfalsdkjfasdjfaksldfjaaskldjfalskdjfasldjfasldjfasldfjalsdjfalsd
Apr 16 20:20:59 localhost.localdomain bash[5174]: jfasdlkfssssssssssssssssssssssssssssssssssssssssssssssssssssssaaaaaaaaaaaaaaaaa
Apr 16 20:20:59 localhost.localdomain bash[5174]: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaafffffffffffffffffffffffffffffffffffffffffffffffffffffffdddddddddddddddddddddddddddd
Apr 16 20:20:59 localhost.localdomain systemd[1]: test.service: Succeeded.
Apr 16 20:20:59 localhost.localdomain systemd[1]: Started test journald line max.
```
And with this pr:
```
Apr 16 20:22:17 localhost.localdomain systemd[1]: Starting test journald line max...
Apr 16 20:22:17 localhost.localdomain bash[6164]: asdkfjasasdfjasdlfajssdjfsdjfskldfjsljfdklsjdflksjdflksjdflsjdflksjdflksdjflsdj
Apr 16 20:22:17 localhost.localdomain bash[6164]: flsdjfasdjfalsjdfalsdkjfasdjfaksldfjaaskldjfalskdjfasldjfasldjfasldfjalsdjfalsd
Apr 16 20:22:17 localhost.localdomain bash[6164]: jfasdlkfssssssssssssssssssssssssssssssssssssssssssssssssssssssaaaaaaaaaaaaaaaaa
Apr 16 20:22:17 localhost.localdomain bash[6164]: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaffffffffffffffffffffffffffffffffffffffffffffffff
Apr 16 20:22:17 localhost.localdomain bash[6164]: fffffffdddddddddddddddddddddddddddd
Apr 16 20:22:17 localhost.localdomain systemd[1]: test.service: Succeeded.
Apr 16 20:22:17 localhost.localdomain systemd[1]: Started test journald line max.
```

**Comment by syyhao1994** [comment]
@bluca PTAL. This PR i have made about 20+ days...

**Comment by syyhao1994** [comment]
@poettering @bluca Please take a look at this.

File: src/journal/journald-stream.c [hunk 1]
```diff
@@ -468,26 +474,41 @@ static int stdout_stream_found(
...
         assert(s);
         assert(p);
 
+        line_max = stdout_stream_line_max(s);
+
         for (;;) {
                 LineBreak line_break;
                 size_t skip, found;
                 char *end1, *end2;
+                size_t tmp_remaining = remaining > line_max ? line_max : remaining;
```
Comment (poettering) [inline src/journal/journald-stream.c]: use MIN()

**Review by poettering — CHANGES_REQUESTED** [review]
looks good, but please use MIN()

File: src/journal/journald-stream.c (hunk 1 above)
Comment (syyhao1994) [reply src/journal/journald-stream.c]: Done! PTAL.

**Comment by syyhao1994** [comment]
@poettering @bluca PTAL.

**Comment by bluca** [comment]
please rebase on latest `main` and force-push, to fix CI errors

**Comment by poettering** [comment]
Lgtm. Lets merge once the ci agrees.

**Comment by NickYang24** [comment]
> 
> 
> please rebase on latest `main` and force-push, to fix CI errors

Thank you for your attention. Rebase done! PTAL.

**Comment by syyhao1994** [comment]
> please rebase on latest `main` and force-push, to fix CI errors

Thank you for you attention. Rebase done! PTAL.

**Comment by syyhao1994** [comment]
@poettering @bluca The CI failure seems nothing to do with this pr. PTAL.

**Pull request merged**